core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -160,6 +160,10 @@ object ByteBufferMessageSet {
             }
         }
       }
+
+      override def close() = {
```

Comment:
We should probably complement/replace the various `compressed.close()` on `case eofe: EOFException` with a finally block that closes `compressed` in case of any exception.